src/api/level.js

```diff
@@ -123,6 +140,269 @@ const UpdateLevel = async (LevelIndex, update) => {
   return updateLevel;
 };
 
+const getLevelIndexesByTags = async (tags, onlyOneMatchIsEnough = false) => {
+  const compareCount = onlyOneMatchIsEnough ? 1 : tags.length;
+
+  let query = null;
+  if (tags.length) {
+    query = `
+      SELECT DISTINCT level_tags.LevelIndex
+      FROM level_tags, level
+      WHERE level_tags.TagIndex IN (${tags.join()})
+	    AND level_tags.LevelIndex = level.LevelIndex
+      GROUP BY level_tags.LevelIndex
+      HAVING COUNT(level_tags.TagIndex) >= ${compareCount};`;
```

Comment:
should maybe use replacements instead of variables inside query
https://sequelize.org/docs/v6/core-concepts/raw-queries/#replacements